File: 1st/app/src/main/java/me/gaojianli/app_playground/MainActivity.java
package me.gaojianli.app_playground;

import android.util.Log;
import android.widget.Button;
import android.widget.CheckBox;
import android.widget.RadioGroup;
import android.widget.TextView;
import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;

public class MainActivity extends AppCompatActivity {

    private static final String TAG = "MainActivity";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Log.d(TAG, "MainActivity Started");
        Button btn1=findViewById(R.id.button);
        final TextView text1=findViewById(R.id.textView);
        btn1.setOnClickListener(v->text1.setText("test"));
        RadioGroup rg1=findViewById(R.id.radioGroup1);
        rg1.setOnCheckedChangeListener((g,index)->{
            int id=g.getCheckedRadioButtonId();
            switch (id){
                case R.id.choose1button:
                    Log.d(TAG, "choose1 is chosen");
                    break;
                case R.id.choose2button:
                    Log.d(TAG, "choose2 is chosen");
                    break;
            }
        });
        CheckBox checkbox=findViewById(R.id.checkBox);
        checkbox.setOnCheckedChangeListener((box,value)->{
            Log.d(TAG, value?"box checked":"box unchecked");
        });
    }
}

File: 2nd/app/src/main/java/me/gaojianli/tiktokmsg/MainActivity.kt
package me.gaojianli.tiktokmsg

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class MainActivity : AppCompatActivity() {
    private var msgList:MutableList<Msg> = ArrayList<Msg>()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        initData()
        val recycleView=findViewById<RecyclerView>(R.id.recyle_view)
        recycleView.layoutManager = LinearLayoutManager(this)
        recycleView.adapter=MsgAdapter(msgList)
    }

    private fun initData(){
        msgList.add(Msg("张三","你好，我是张三",R.mipmap.zhang3))
        msgList.add(Msg("李四","你好，我是李四",R.mipmap.li4))
        msgList.add(Msg("王二麻子","你好，我是王二麻子",R.mipmap.wang2mazi))
        msgList.add(Msg("赵五","你好，我是赵五",R.mipmap.zhao5))
    }
}
